drivers/usb/usbstor/pdo.c

```diff
@@ -437,7 +437,7 @@ USBSTOR_PdoHandleQueryInstanceId(
 {
     PPDO_DEVICE_EXTENSION PDODeviceExtension;
     PFDO_DEVICE_EXTENSION FDODeviceExtension;
-    WCHAR Buffer[100];
+    WCHAR Buffer[128];
```

Comment:
Any reason it's still a hardcoded value? Maybe should be a constant? What the spec says?

Reply:
It's basically MAXIMUM_USB_STRING_LENGTH divided by 2 (since sizeof(WCHAR) is 2 bytes). Could put there something like `MAXIMUM_USB_STRING_LENGTH/sizeof(WCHAR)` if preferable.

Reply:
Interestingly, in the PSDK there are two fields where it's WCHAR array of that size (i.e. not divided by two) : https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/psdk/usbrpmif.h;hb=deee8aaec17fe841054cd5e8d4936e0279a3f47d#l38

Reply:
Not sure why. I did this based on comment [here](https://github.com/mcuee/libusbk/blob/7318e61d6eaf22327fea74e46dddc1bf82e0e248/libusbK/includes/libusbk.h#L463). Also [this answer](https://stackoverflow.com/a/7283272) suggests that the max string descriptor length is 255 bytes.

Reply:
I think it would then be more readable to do your suggestion @archeYR i.e. `MAXIMUM.... / sizeof(WCHAR) + 1`. This would help clearly indicate what the length value is.

Reply:
Actually I realized later on that this should also account the LUN number... Anyway I think I will just use disean's implementation for this.